rocketpy/sensors/gyroscope.py

```diff
@@ -0,0 +1,337 @@
+            temperature_scale_factor=temperature_scale_factor,
+            cross_axis_sensitivity=cross_axis_sensitivity,
+            name=name,
+        )
+        self.acceleration_sensitivity = self._vectorize_input(
+            acceleration_sensitivity, "acceleration_sensitivity"
+        )
+        self.prints = _GyroscopePrints(self)
+
+    def measure(self, t, u, u_dot, relative_position, *args):
+        """Measure the angular velocity of the rocket
+
+        Parameters
+        ----------
+        t : float
+            Time at which the measurement is taken
+        u : list
+            State vector of the rocket
+        u_dot : list
+            Time derivative of the state vector of the rocket
+        relative_position : Vector
+            Vector from the rocket's center of mass to the sensor
+        """
+        # Angular velocity of the rocket in the rocket frame
+        omega = Vector(u[10:13])
+
+        # Transform to sensor frame
+        inertial_to_sensor = self._total_rotation_matrix @ Matrix.transformation(
+            u[6:10]
+        )
+        W = inertial_to_sensor @ omega
+
+        # Apply noise + bias
+        # Apply noise + bias and quatize
+        W = self.apply_noise(W)
+        W = self.apply_temperature_drift(W)
+
+        # Apply acceleration sensitivity
+        if self.acceleration_sensitivity != Vector.zeros():
+            W += self.apply_acceleration_sensitivity(
+                omega, u_dot, relative_position, inertial_to_sensor
+            )
+
+        W = self.quantize(W)
+
+        self.measurement = tuple([*W])
+        self._save_data((t, *W))
+
+    def apply_acceleration_sensitivity(
+        self, omega, u_dot, relative_position, rotation_matrix
+    ):
+        """
+        Apply acceleration sensitivity to the sensor measurement
+
+        Parameters
+        ----------
+        omega : Vector
+            The angular velocity to apply acceleration sensitivity to
+        u_dot : list
+            The time derivative of the state vector
+        relative_position : Vector
+            The vector from the rocket's center of mass to the sensor
+        rotation_matrix : Matrix
+            The rotation matrix from the rocket frame to the sensor frame
+
+        Returns
+        -------
+        Vector
+            The angular velocity with the acceleration sensitivity applied
+        """
+        # Linear acceleration of rocket cdm in inertial frame
+        a_I = Vector(u_dot[3:6])
+
+        # Angular velocity and accel of rocket
+        omega_dot = Vector(u_dot[10:13])
+
+        # Acceleration felt in sensor
+        A = (
+            a_I
+            + Vector.cross(omega_dot, relative_position)
+            + Vector.cross(omega, Vector.cross(omega, relative_position))
+        )
+        # Transform to sensor frame
+        A = rotation_matrix @ A
+
+        return self.acceleration_sensitivity & A
+
+    def export_measured_data(self, filename, format="csv"):
```

Comment:
For me this implementation of export_measured_data should be done in the parent class, otherwise you will have to re-implement it in every child class. The code in the Gyroscope and Accelerometer classes is almost identical, so it should be in the parent class.

Something  like this:

```python
def export_measured_data(self, filename, format="csv", data_labels=("t", "x", "y", "z")):
        """
        Export the measured values to a file

        Parameters
        ----------
        filename : str
            Name of the file to export the values to
        format : str
            Format of the file to export the values to. Options are "csv" and
            "json". Default is "csv".
        data_labels : tuple
            Tuple of strings representing the labels for the data columns

        Returns
        -------
        None
        """
        if format.lower() not in ["json", "csv"]:
            raise ValueError("Invalid format")
        
        if format.lower() == "csv":
            # if sensor has been added multiple times to the simulated rocket
            if isinstance(self.measured_data[0], list):
                print("Data saved to", end=" ")
                for i, data in enumerate(self.measured_data):
                    with open(filename + f"_{i+1}", "w") as f:
                        f.write(",".join(data_labels) + "\n")
                        for entry in data:
                            f.write(",".join(map(str, entry)) + "\n")
                    print(filename + f"_{i+1},", end=" ")
            else:
                with open(filename, "w") as f:
                    f.write(",".join(data_labels) + "\n")
                    for entry in self.measured_data:
                        f.write(",".join(map(str, entry)) + "\n")
                print(f"Data saved to {filename}")
            return
        
        if format.lower() == "json":
            if isinstance(self.measured_data[0], list):
                print("Data saved to", end=" ")
                for i, data in enumerate(self.measured_data):
                    data_dict = {label: [] for label in data_labels}
                    for entry in data:
                        for label, value in zip(data_labels, entry):
                            data_dict[label].append(value)
                    with open(filename + f"_{i+1}", "w") as f:
                        json.dump(data_dict, f)
                    print(filename + f"_{i+1},", end=" ")
            else:
                data_dict = {label: [] for label in data_labels}
                for entry in self.measured_data:
                    for label, value in zip(data_labels, entry):
                        data_dict[label].append(value)
                with open(filename, "w") as f:
                    json.dump(data_dict, f)
                print(f"Data saved to {filename}")
            return
```

Reply:
I'll leave this change to the next sensors PR. It will be easier to do it there

Reply:
> I'll leave this change to the next sensors PR. It will be easier to do it there

Ok please don't forget it.